include/circt/Dialect/SV/SVStatements.td

```diff
@@ -720,3 +720,28 @@ def DepositOp : SVOp<"nonstandard.deposit",
   let results = (outs);
   let assemblyFormat = "$dest `,` $src  attr-dict `:` qualified(type($src))";
 }
+
+
+//===----------------------------------------------------------------------===//
+// Memory loading system tasks
+//===----------------------------------------------------------------------===//
+
+def MemBaseBin: I32EnumAttrCase<"MemBaseBin", 0>;
+def MemBaseHex: I32EnumAttrCase<"MemBaseHex", 1>;
+def MemBaseTypeAttr :
+    I32EnumAttr<"MemBaseTypeAttr", "the numeric base of a memory file",
+                [MemBaseBin, MemBaseHex]>;
+
+def ReadMemOp : SVOp<"readmem"> {
```

Comment:
Is it possible to use `readmem` in non-procedural region? If no, can we add `ProceduralOp` trait here?

Reply:
While this seems entirely reasonable, I was uncertain (and did not experiment) about if it was actually disallowed. The spec states that:

> These tasks can be executed at any time during simulation.

I interpreted that as "we can't restrict this to `ProceduralOp`" even though any sane usage of it would be. This perhaps motivates restricting this to a procedural region. 🤔

Reply:
I interpreted "at any time during simulation" as "we can use readmem in any procedure.". At least verilator complained about readmem being putted on non-procedural blocks.

Reply:
Thanks for checking! I'll add this.